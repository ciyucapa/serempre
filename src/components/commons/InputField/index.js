import React from 'react';
import PropTypes from 'prop-types';

const InputField = (props) => {
    const {error} = props;
    return (
        <div style={{padding: 10}}>
            <input {...props} />
            {error && (
                <div style={{color: '#FF0000'}}>{error}</div>
            )}
        </div>
    );
};

InputField.propTypes = {
    value: PropTypes.string,
    onChange: PropTypes.func,
    placeholder: PropTypes.string,
    error: PropTypes.string,
};

InputField.defaultProps = {
    value: '',
    onChange: '',
    placeholder: '',
    error: '',
};

export default InputField;
