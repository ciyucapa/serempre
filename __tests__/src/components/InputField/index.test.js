import React from 'react';
import renderer from 'react-test-renderer';

import InputField from '../../../../src/components/commons/InputField';

describe('InputField', () => {
    it('renders correctly', () => {
        const props = {
            value: 'test',
            onChange: () => {},
            placeholder: 'test placeholder',
            error: '',
        };

        const tree = renderer
            .create(<InputField {...props} />)
            .toJSON();

        expect(tree).toMatchSnapshot();
    });
});
