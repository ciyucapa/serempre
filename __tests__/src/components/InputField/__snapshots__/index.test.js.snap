// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputField renders correctly 1`] = `
<div
  className="sc-bdnylx lfgfey"
>
  <input
    className="sc-gtssRu glhVaN"
    onChange={[Function]}
    placeholder="test placeholder"
    value="test"
  />
  
</div>
`;
